scripts/nayduck.py

```diff
@@ -203,6 +207,24 @@ def main():
         'tests': list(tests)
     }
 
+    if args.run_locally:
+        for test in post["tests"]:
+            # TODO: use parses similar to nayduck, where is it?
+            fields = test.split(" ")
+
+            if fields[0] == "expensive":
+                print("running expensive test %s" % test)
+                # timeout field not implemented
+                if fields[1].startswith("--timeout="):
+                    del fields[1]
+
+                package = fields[1]
+                params = test.split('::')[-1].split(" ")
+                subprocess.check_call(
+                    ['cargo', 'test', '-p', package, '--tests'] + params)
+            else:
+                print("ignoring don't know how to run locally %s" % test)
+
```

Comment:
At this point it’s time to split this into multiple functions. One for local_run (with this code) and the other for remote run (with code for getting the auth code, post data, and the `while True` loop below).